File: Backend/src/main/java/com/airecruitment/interview/service/InterviewService.java
package com.airecruitment.interview.service;

import com.airecruitment.interview.dto.InterviewEvaluationResponse;
import com.airecruitment.interview.dto.InterviewQuestionResponse;
import com.airecruitment.interview.dto.InterviewQuestionsResponse;
import com.airecruitment.interview.dto.InterviewResultResponse;

import java.util.List;
import com.airecruitment.interview.dto.InterviewListResponse;
import org.springframework.web.multipart.MultipartFile;

public interface InterviewService {

    InterviewQuestionsResponse generateQuestions(
            Long jobId,
            Long candidateId,
            Long resumeId
    );

    InterviewEvaluationResponse submitAnswer(
            Long candidateId,
            Long questionId,
            String answer
    );

    InterviewEvaluationResponse submitVoiceAnswer(
            Long candidateId,
            Long questionId,
            MultipartFile audio
    );

    InterviewResultResponse terminateInterview(
            Long candidateId,
            Long jobId,
            Integer violationCount,
            String reason
    );

    List<InterviewQuestionResponse> getQuestions(
            Long candidateId,
            Long jobId,
            Long resumeId
    );

    InterviewResultResponse getInterviewResult(
            Long candidateId,
            Long jobId
    );

    List<InterviewListResponse> getMyInterviews(Long candidateId);
}

File: Frontend/Frontend/src/services/interviewService.js
import api from "./api";

const getMyInterviews = async (candidateId) => {
  const response = await api.get(
    `/interview/candidates/${candidateId}/interviews`
  );

  return response.data;
};

const generateQuestions = async (
  jobId,
  candidateId,
  resumeId
) => {
  const response = await api.post(
    `/interview/jobs/${jobId}/candidates/${candidateId}/resumes/${resumeId}/questions`
  );

  return response.data;
};

const getQuestions = async (
  candidateId,
  jobId,
  resumeId
) => {
  const response = await api.get(
    `/interview/candidates/${candidateId}/jobs/${jobId}/resumes/${resumeId}/questions`
  );

  return response.data;
};

const submitAnswer = async (
  candidateId,
  questionId,
  answer
) => {
  const response = await api.post(
    `/interview/candidates/${candidateId}/questions/${questionId}/answer`,
    {
      answer,
    }
  );

  return response.data;
};

const submitVoiceAnswer = async (
  candidateId,
  questionId,
  audioBlob
) => {

  const formData =
    new FormData();

  formData.append(
    "audio",
    audioBlob,
    "voice-answer.webm"
  );

  const response =
    await api.post(
      `/interview/candidates/${candidateId}/questions/${questionId}/voice-answer`,
      formData
    );

  return response.data;
};

const terminateInterview = async (
  candidateId,
  jobId,
  violationCount,
  reason
) => {
  const response = await api.post(
    `/interview/candidates/${candidateId}/jobs/${jobId}/terminate`,
    {
      violationCount,
      reason,
    }
  );

  return response.data;
};

const getInterviewResult = async (
  candidateId,
  jobId
) => {
  const response = await api.get(
    `/interview/candidates/${candidateId}/jobs/${jobId}/result`
  );

  return response.data;
};
export default {
  generateQuestions,
  getQuestions,
  submitAnswer,
  submitVoiceAnswer,
  getInterviewResult,
  getMyInterviews,
  terminateInterview,
};